aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -2841,20 +2841,33 @@ static void apply_lu_solve_batched_magma(const Tensor& b, const Tensor& lu, cons
 #endif
 }
 
-static void lu_solve_magma(const Tensor& b, const Tensor& lu, const Tensor& pivots) {
```

Comment:
Could you please split the original `lu_solve_magma` function into `lu_solve_looped_magma` (that dispatches based on dtype to `apply_lu_solve_looped_magma`) and `lu_solve_batched_magma` (that dispatches based on dtype to `apply_lu_solve_batched_magma`) and define a new function `lu_solve` that calls cusolver/cublas/magma looped/batched variants. Here the mix of `lu_solve_*` and `apply_lu_solve_*` with `AT_DISPATCH` makes things look complicated.